app/pages/admin/project-status.jsx

```diff
@@ -125,16 +125,19 @@ class ProjectStatus extends Component {
   }
 
   handleFeaturedProjectChange({ target }) {
-    const { featured, project } = this.state;
-    if (featured) {
-      featured.update({ featured: false });
-      featured.save()
-        .then(() => this.setState({ featured }))
+    const { featuredProject, project } = this.state;
+    project.update({ featured: target.checked });
+    if (featuredProject) {
+      return featuredProject.update({ featured: false }).save()
+        .then(() => project.save())
```

Comment:
If this save (or the previous one) throws an error, then `state.featuredProject.featured` is false, and `project.featured` is true, but I think the checkbox might still be active. I'm not sure.

I'm not sure how big a deal error checking is, or how the UI is intended to react if Panoptes is having problems. I don't think any of the other admin form elements have great error handling either.

Reply:
I'm going to catch the error before `setState`, which should help with your point. The UI just consists of the error message being displayed next to the checkbox. Also when Panoptes has problems, e.g. screenshot below
<img width="740" alt="screen shot 2018-04-03 at 10 40 54" src="https://user-images.githubusercontent.com/8989606/38241998-8ffef9d0-372b-11e8-92bd-62a03eda5179.png">

Reply:
Just waiting for checks to pass, before merging.

Reply:
Remember that a then block after a catch always runs. In the event of an error, it will receive the return value of the catch block.

Reply:
The latest commit sets `state.featuredProject`, even after an error occurs. I'm not sure what it sets it to after an error, so the component could get into a weird state there. No big deal if refreshing the page clears the error.

Reply:
It does.

Reply:
But I'll remove the latest commit, just because it's confusing/has the wrong message.

Reply:
Yeah, if the code is going to continue and set state after an error, then the error handler should probably reverse any unsaved changes.